cypress/integration/core/date-picker.spec.ts

```diff
@@ -400,22 +410,27 @@ describe('Date Picker', () => {
       });
   });
 
-  it('should emit changed event with error on invalid character paste', () => {
+  it('should not allow invalid characters to be pasted', () => {
     const details = [];
     cy.get('dso-date-picker').then(datePicker => {
-      datePicker.get(0).addEventListener('dateChange', (event: CustomEvent) => details.push(event.detail))
+      datePicker.get(0).addEventListener('dateChange', (event: CustomEvent) => details.push(event.detail));
     });
 
-    cy
+    const input = cy
       .get('dso-date-picker')
-      .find('input.dso-date__input')
+      .find('input.dso-date__input');
+
+    input
       .invoke('val', 'zzz')
       .trigger('input')
       .then(() => {
-        console.log(details)
-        expect(details[details.length - 1].error).equal('invalid');
-        expect(details[details.length - 1].value).equal('zzz');
-        expect(details[details.length - 1].valueAsDate).undefined;
+        input.then($input => {
+          expect($input.val()).equal('');
+          expect(details.length).equal(1);
+          expect(details[details.length - 1].error).undefined;
+          expect(details[details.length - 1].value).equal('');
+          expect(details[details.length - 1].valueAsDate).equal(undefined);
+        });
```

Comment:
Wat denk je van:

```ts
  it('should not allow invalid characters to be pasted', () => {
    const details = [];
    cy.get('dso-date-picker').then(datePicker => {
      datePicker.get(0).addEventListener('dateChange', (event: CustomEvent) => details.push(event.detail));
    });

    cy
      .get('dso-date-picker input.dso-date__input')
      .type('zzz')
      .should('have.value', '')

    cy.wrap(details)
      .should('have.length', 1)
      .invoke('at', 0)
      .should('deep.equal', { component: 'dso-date-picker', value: '', valueAsDate: undefined })
  });
```

?

Reply:
Ja mooi, alleen beetje jammer dat cypress niet de waarde geeft van wat hij deep-equaled.
<img width="422" alt="image" src="https://user-images.githubusercontent.com/21218284/180382483-e0ad0cc5-70dd-4809-bd10-4232f6470f8e.png">

Verder heb ik `.type()` niet overgenomen, omdat dit een paste-action test is.

Reply:
Ja dat is idd jammer..

Over `.type()`: goed punt.